### PR TITLE
LibWeb: Support background attribute on table elements

### LINKED ISSUE #19806: [Feature Request/Ladybird]: support background attribute in tr/td
I guess it could be mapped to css `background/background-image` properties.

See https://amigaworld.net:

Chrome:
Capture d’écran 2023-07-05 à 11 13 02

Ladybird:
Capture d’écran 2023-07-05 à 11 13 52

### PR DESCRIPTION
**Pull request by warpdesign**
Fixes #19806

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/HTMLTableCellElement.cpp [hunk 1]
```diff
@@ -61,6 +63,9 @@ void HTMLTableCellElement::apply_presentational_hints(CSS::StyleProperties& styl
             if (auto parsed_value = parse_nonzero_dimension_value(value))
                 style.set_property(CSS::PropertyID::Height, parsed_value.release_nonnull());
             return;
+        } else if (name == HTML::AttributeNames::background) {
+            style.set_property(CSS::PropertyID::BackgroundImage, CSS::ImageStyleValue::create(document().parse_url(value)).release_value_but_fixme_should_propagate_errors());
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/HTML/HTMLTableCellElement.cpp:67]: We should check if the parsed url `is_valid()` before trying to make a StyleValue for it.

File: Userland/Libraries/LibWeb/HTML/HTMLTableRowElement.cpp [hunk 2]
```diff
@@ -43,6 +45,9 @@ void HTMLTableRowElement::apply_presentational_hints(CSS::StyleProperties& style
             if (color.has_value())
                 style.set_property(CSS::PropertyID::BackgroundColor, CSS::ColorStyleValue::create(color.value()).release_value_but_fixme_should_propagate_errors());
             return;
+        } else if (name == HTML::AttributeNames::background) {
+            style.set_property(CSS::PropertyID::BackgroundImage, CSS::ImageStyleValue::create(document().parse_url(value)).release_value_but_fixme_should_propagate_errors());
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/HTML/HTMLTableRowElement.cpp:49]: Same here of course.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
For reference, you don't need to create an issue for things you're already working on. Create an issue if you find a problem and aren't going to work on it, otherwise just submit the PR.

**Comment by warpdesign** [comment]
> For reference, you don't need to create an issue for things you're already working on. Create an issue if you find a problem and aren't going to work on it, otherwise just submit the PR.

I created the issue before working on it: didn't know if I could pull this through when i opened it :)

**Comment by warpdesign** [comment]
Should be good now.

**Review by AtkinsSJ — APPROVED** [review]
Looks good! That Amiga site will be looking right in no time.

**Pull request merged**